### PR TITLE
Fixed QC display syntax for Docker users

### LINKED ISSUE #2585: QC open command should be hidden if users don't have access to local browser
For users running SCT under Docker in Windows via Xming, when launching a process with the flag `-qc`, the command will output a syntax to open this html report, using the default Linux program: `xdg-open`. The problem is that, even though the program is installed, it cannot find a default browser because there is no browser installed in the SCT Docker container, yielding the following error: 

~~~
$open_cmd index.html
Couldn't find a suitable web browser!
Set the BROWSER environment variable to your desired browser.
Warning: program returned non-zero exit code #1
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: www-browser: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: links2: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: elinks: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: links: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: lynx: not found
/usr/bin/xdg-open: 851: /usr/bin/xdg-open: w3m: not found
xdg-open: no method available for opening 'index.html'
~~~

One solution would be to look interrogate the variable `BROWSER`, and if not defined, change the message for "Double click on the file: PATH/index.html"

### PR DESCRIPTION
**Pull request by lrouhier**
Two issues appeared regarding the use of SCT in our released docker container : 

#2585 : For docker user, they can't copy-paste the `xdg-open` command because no browser is set. If the variable `BROWSER` is not set for the Linux user, this command won't work as well. For now, we just modified the command given to the user if he uses Linux without defining `$BROWSER`. 

#2587: Fsleyes plugin is not working for docker users. After investigations, the crash is caused by the deletion  [here](https://github.com/neuropoly/spinalcordtoolbox/blob/0ab51b7d97eaecd539ac11b6fdb07a6ea6fb7548/contrib/fsl_integration/sct_plugin.py#L35) of 'PYTHONHOME' and 'PYTHONPATH' that are not defined. I added a test to avoid this situation. The plugin seems to be responsive now.  Fixed by #2581 

However, it does not solve the warning problem reported also in the issue. It might come from the installation of fsleyes in the docker container which is currently investigated (WIP)

intent to fix #2585, #2587

### CODE REVIEW

File: spinalcordtoolbox/reports/qc.py [hunk 1]
```diff
@@ -561,7 +561,12 @@ def layout(qslice):
     try:
         from sys import platform as _platform
         if _platform == "linux" or _platform == "linux2":
-            sct.printv('xdg-open "{}/index.html"'.format(path_qc), type='info')
+            try:  
+                os.environ["BROWSER"]
+                sct.printv('xdg-open "{}/index.html"'.format(path_qc), type='info')
+            except KeyError:
```
Comment (jcohenadad) [inline spinalcordtoolbox/reports/qc.py]: could you please comment in the code when this KeyError is supposed to happen?

Comment (jcohenadad) [reply spinalcordtoolbox/reports/qc.py]: also, are you 100% sure that in ALL linux platforms, this BROWSER variable exists if a browser is accessible?

Comment (lrouhier) [reply spinalcordtoolbox/reports/qc.py]: Xdg-open was asking for it, but now I am not sure that it is automatically set. 
The error is thrown if the variable is not previously declared or empty. 
I see two option : 
* there is webbrowser python package that can do the test without going through the $BROWSER variable.
* We can set a $DOCKER env variable for the docker container and use the same Key error.

Comment (jcohenadad) [reply spinalcordtoolbox/reports/qc.py]: > We can set a $DOCKER env variable for the docker container and use the same Key error.

I like this approach.

Comment (jcohenadad) [reply spinalcordtoolbox/reports/qc.py]: it seems a bit cleaner somehow...

**Comment by jcohenadad** [comment]
@lrouhier before we merge: is the DOCKER variable now set in the new Dockerfile? if so can you please point to the modified code?

**Comment by lrouhier** [comment]
> @lrouhier before we merge: is the DOCKER variable now set in the new Dockerfile? if so can you please point to the modified code?

It is on the PR https://github.com/neuropoly/sct_docker/pull/51, here is the [link](https://github.com/neuropoly/sct_docker/blob/1a2de7a8f0602c2087ba27d5bb5ad91b8150accb/sct_docker.py#L84) to the exact line.  Do you want me to add something as a comment inside the QC file ?

**Comment by jcohenadad** [comment]
>  inside the QC file

which file are you talking about?

**Comment by lrouhier** [comment]
Sorry I meant in the `qc.py` file. For futur reference if it is needed.

**Comment by jcohenadad** [comment]
ah! yes, it would be nice. Thanks!

**Pull request merged**